mithril-aggregator/src/runtime/runner.rs

```diff
@@ -264,6 +276,54 @@ impl AggregatorRunnerTrait for AggregatorRunner {
             .await
     }
 
+    async fn is_slave_aggregator_at_same_epoch_as_master(
+        &self,
+        time_point: &TimePoint,
+    ) -> StdResult<bool> {
+        let is_slave_aggregator_at_same_epoch_as_master = if let Some(master_epoch_settings) = self
+            .dependencies
+            .master_aggregator_client
+            .retrieve_epoch_settings()
+            .await?
+        {
+            time_point.epoch == master_epoch_settings.epoch
+        } else {
+            false
+        };
+
+        Ok(is_slave_aggregator_at_same_epoch_as_master)
+    }
+
+    async fn synchronize_slave_aggregator_signer_registration(&self) -> StdResult<()> {
+        let master_epoch_settings = self
+            .dependencies
+            .master_aggregator_client
+            .retrieve_epoch_settings()
+            .await?
+            .ok_or(anyhow::anyhow!(
+                "Master aggregator did not return any epoch settings"
+            ))?;
+
+        let registration_epoch = master_epoch_settings
+            .epoch
+            .offset_to_master_synchronization_epoch();
+        let next_signers = master_epoch_settings.next_signers;
+        let stake_distribution = self
+            .dependencies
+            .stake_store
+            .get_stakes(registration_epoch)
+            .await?
+            .ok_or(anyhow::anyhow!(
+                "Slave aggregator did not return any stake distribution"
+            ))?;
+        self.dependencies
+            .signer_synchronizer
+            .synchronize_signers(registration_epoch, &next_signers, &stake_distribution)
+            .await?;
+
+        Ok(())
+    }
```

Comment:
Would it be better to implement those two behaviours in the synchroniser directly ?